subsys/bluetooth/common/rpa.h

```diff
@@ -0,0 +1,21 @@
+/* rpa.h - Bluetooth Resolvable Private Addresses (RPA) generation and
+ * resolution
+ */
+
+/*
+ * Copyright (c) 2017 Nordic Semiconductor ASA
+ * Copyright (c) 2015-2016 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+
+#if defined(CONFIG_BLUETOOTH_SMP)
+bool bt_rpa_irk_matches(const u8_t irk[16], const bt_addr_t *addr);
+#endif
+
```

Comment:
I don't think you need these #ifdef guards here. If the c-file that's making these calls is conditionally compiled based on the same condition, then that's already fine. For c-files which may or may not have the condition fulfilled you can use things like the following (though I suspect the calls are already behind preprocessor guards, so probably nothing is needed):

 if (IS_ENABLED(CONFIG_BLUETOOTH_SMP)) {
         bt_rpa_irk_matches(...);
 }